hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -2666,6 +2666,17 @@ public void deleteObjectTagging(OmKeyArgs args) throws IOException {
     handleError(omResponse);
   }
 
+  @Override
+  public void compactOMDB(String columnFamily) throws IOException {
+    OzoneManagerProtocolProtos.CompactRequest compactRequest =
+        OzoneManagerProtocolProtos.CompactRequest.newBuilder()
+            .setColumnFamily(columnFamily)
+            .build();
+    OMRequest omRequest = createOMRequest(Type.CompactDB)
+        .setCompactRequest(compactRequest).build();
+    handleError(submitRequest(omRequest));
```

Comment:
Digging into this, it looks like it will only go to the [leader](https://github.com/apache/ozone/blob/2f3150ab68183b8db61ee758c793fd4c1e1ee594/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/Hadoop3OmTransport.java#L107-L108). For this command, I think we would like it to go to each OM individually as specified when the command is given. We will probably need to look at re-using/refactoring the `ReconfigureProtocolClientSideTranslatorPB` to work more generally for debug/repair commands that need to be run online and not go through Ratis. Otherwise we may end up needing a lot of new/duplicate code to accomplish this.

Reply:
agreed. Good catch.

Reply:
Yes, good find I didn't know we had this as well. Adding this to the admin protocol looks like a better option.